test/e2e_node/util.go

```diff
@@ -85,6 +86,34 @@ func getCurrentKubeletConfig() (*componentconfig.KubeletConfiguration, error) {
 	return kubeCfg, nil
 }
 
+// Convenience method to set the evictionHard threshold during the current context.
+func tempSetEvictionHard(f *framework.Framework, evictionHard string) {
+	tempSetCurrentKubeletConfig(f, func(initialConfig *componentconfig.KubeletConfiguration) {
+		initialConfig.EvictionHard = evictionHard
+	})
+}
+
+// Must be called within a Context. Allows the function to modify the KubeletConfiguration during the BeforeEach of the context.
+// The change is reverted in the AfterEach of the context.
+func tempSetCurrentKubeletConfig(f *framework.Framework, updateFunction func(initialConfig *componentconfig.KubeletConfiguration)) {
+	var oldCfg *componentconfig.KubeletConfiguration
+	BeforeEach(func() {
+		oldCfg, err := getCurrentKubeletConfig()
```

Comment:
I suspect this is going to create a new `oldCfg` variable shadowing the shared one, so that the `oldCfg` in `AfterEach` will always be nil. Can you confirm either way?